stl/src/cthread.cpp

```diff
@@ -87,7 +87,7 @@ void _Thrd_yield() { // surrender remainder of timeslice
 
 // TRANSITION, ABI: _Thrd_equal() is preserved for binary compatibility
 _CRTIMP2_PURE int _Thrd_equal(_Thrd_t thr0, _Thrd_t thr1) { // return 1 if thr0 and thr1 identify same thread
-    return thr0._Id == thr1._Id;
+    return static_cast<int>(thr0._Id == thr1._Id);

```

Comment:
Why the static cast here?
Also don't bool -> int have an implicit conversion operator?

Reply:
To emphasize this is to return an int for C ABI purposes (even though in practice if we returned as a bool, it shouldn't cause issues)

Reply:
@AraHaan: Yes, there exists implicit conversions from `bool` to `int`. From the C++ Working Draft, [§7.3.7.6](https://eel.is/c++draft/conv#prom-6):

> A prvalue of type `bool` can be converted to a prvalue of type `int`, with `false` becoming zero and `true` becoming one.

With that said, this seems more like a stylistic choice. If it was written without a `static_cast` before, then that's probably how they would want it, for consistency. I would advise checking out what is done in this scenario in other situations.

If the goal is to emphasize that the `int` return type is for C ABI, then perhaps the better idea would be to change the leading comment to display, e.g.,

```C++
// TRANSITION, ABI: _Thrd_equal() is preserved for binary compatibility, returns int for C ABI
_CRTIMP2_PURE int _Thrd_equal(_Thrd_t thr0, _Thrd_t thr1) { // return 1 if thr0 and thr1 identify same thread
[. . .]
```